Completed Потренируем StreamAPI 1

elka1907 opened the pull request:
ошибки в наименовании методов в примере с подсказками:
1. в примере:  а)  System.out.println("Строки, содержащие 'an': " + ListOperations.**filterStringsContainingSubstring**(strings, "an"));
б)  System.out.println("Длины строк: " + ListOperations.**convertToLengths**(strings));
3. в подсказках:
а)  public static List<String> **filterBySubstring**(List<String> strings, String substring) {
б)  public static List<Integer> **mapToLengths**(List<String> strings) {

elka1907 committed:
Completed Потренируем StreamAPI 1

MarselYsup commented on src/main/java/school/faang/bjs2_88245/ListOperations.java:
```diff
@@ -0,0 +1,65 @@
+                .orElseThrow(() -> new IllegalArgumentException("Cannot calculate average of empty list"));
+    }
+
+    public static long countStringsStartingWith(List<String> strings, char c) {
```
с - непрозрачное название для параметра метода, лучше назови symbol

elka1907 replied:
Done

MarselYsup commented on src/main/java/school/faang/bjs2_88245/ListOperations.java:
```diff
@@ -0,0 +1,65 @@
+
+    public static int findMinGreaterThan(List<Integer> numbers, int threshold) {
+        return numbers.stream()
+                .filter(x -> x > threshold)
```
когда пишешь лямбду также не используй просто параметр x, a, b и тд
представь если лямбда будет большой и в реальной разработке будет сложно понимать, что такое x

elka1907 replied:
Done, mercy

MarselYsup reviewed (commented):
Поправь нейминг, остальное хорошо 👍

elka1907 committed:
Скорректировала:
1. char c на char symbol
2. .filter(x -> x % 2 == 0) на .filter(number -> number % 2 == 0)
3. .filter(s -> !s.isEmpty() && s.charAt(0) == c) на  .filter(string -> !string.isEmpty() && string.charAt(0) == symbol)
4. .filter(s -> s.contains(substring)) на .filter(string -> string.contains(substring))
5. .filter(x -> x > threshold) на .filter(number -> number > threshold)
